src/TextInput/TextInput.tsx

```diff
@@ -28,14 +28,26 @@ type Props = {
   error?: boolean
   /** error text message */
   errorMsg?: string
-  /** onChange listener */
-  onChange: (e: string) => void
+  /** onBlur listener */
+  onBlur?: (e: FormEvent<HTMLInputElement>) => void
   /** onChange listener */
   trailingIcon?: string
   /** Disabled flag */
   disabled?: boolean
 }
 
+type TruncateProps =
+  | {
+      onChange: (e: string) => void
+      onInputChange?: (e: FormEvent<HTMLInputElement>) => void
+    }
+  | {
+      onChange?: (e: string) => void
+      onInputChange: (e: FormEvent<HTMLInputElement>) => void
+    }
```

Comment:
Wondering if it's worth adding a comment to explain what's the difference and how both are used